src/devices/Common/Iot/Device/Common/ValueArray.cs

```diff
@@ -277,11 +290,43 @@ public void RemoveAt(int index)
         {
             get
             {
-                return AsSpan()[index];
+                if ((uint)index >= (uint)_count)
+                {
+                    throw new IndexOutOfRangeException();
```

Comment:
nit: this can use a throw helper to improve the generated code a bit, see [this page](https://docs.microsoft.com/en-us/windows/communitytoolkit/developer-tools/throwhelper#technical-details) for details.

Reply:
Ok, but I think that should be done in a separate commit over (more or less) the whole repo.